ANGLE translator fuzzer runs with resources incompatible with the output

kkinnunen-apple opened the pull request:
#### 0f6c7e4175f921ca35157e448b1708d468dffe35
<pre>
ANGLE translator fuzzer runs with resources incompatible with the output
<a href="https://bugs.webkit.org/show_bug.cgi?id=283406">https://bugs.webkit.org/show_bug.cgi?id=283406</a>
<a href="https://rdar.apple.com/140263046">rdar://140263046</a>

Reviewed by Tadeu Zagallo.

Filter the resources based on the output type. As an example, MSL output
does not support ARB_texture_rectangle, so fuzzer finding
texture2DRect() asserts needlessly. The backend does not enable this.

Only filter MSL extensions correctly for now.

* Source/ThirdParty/ANGLE/WebKit/TranslatorFuzzer.cpp:
(sh::initializeTranslators):
* Source/ThirdParty/ANGLE/WebKit/TranslatorFuzzerDumpTestCase.cpp:
* Source/ThirdParty/ANGLE/WebKit/TranslatorFuzzerSupport.h:

Canonical link: <a href="https://commits.webkit.org/286907@main">https://commits.webkit.org/286907@main</a>
</pre>

tadeuzagallo commented on Source/ThirdParty/ANGLE/WebKit/TranslatorFuzzerSupport.h:
```diff
@@ -113,6 +113,64 @@
 void filterOptions(ShShaderOutput output, ShCompileOptions& options);
 ShShaderOutput resolveShaderOutput(ShShaderOutput output);
 
+#define FOR_EACH_SH_BUILT_IN_RESOURCES_EXTENSION_OPTION(MACRO) \
```
would it make sense for `any` and `msl` to be parameters of this macro?

kkinnunen-apple replied:
> would it make sense for `any` and `msl` to be parameters of this macro?

No, not really. If there's 77 backends, we cannot have 77 parameters. I just added msl, since I didn't want to go through other code, but that's the pattern for all other backends.

kkinnunen-apple committed:
ANGLE translator fuzzer runs with resources incompatible with the output
https://bugs.webkit.org/show_bug.cgi?id=283406
rdar://140263046

Reviewed by Tadeu Zagallo.

Filter the resources based on the output type. As an example, MSL output
does not support ARB_texture_rectangle, so fuzzer finding
texture2DRect() asserts needlessly. The backend does not enable this.

Only filter MSL extensions correctly for now.

* Source/ThirdParty/ANGLE/WebKit/TranslatorFuzzer.cpp:
(sh::initializeTranslators):
* Source/ThirdParty/ANGLE/WebKit/TranslatorFuzzerDumpTestCase.cpp:
* Source/ThirdParty/ANGLE/WebKit/TranslatorFuzzerSupport.h:

Canonical link: https://commits.webkit.org/286907@main
